src/components/swap/SwapModalFooter.tsx

```diff
@@ -121,46 +175,135 @@ export default function SwapModalFooter({
   swapQuoteReceivedDate: Date | undefined
   fiatValueInput: { data?: number; isLoading: boolean }
   fiatValueOutput: { data?: number; isLoading: boolean }
+  shouldLogModalCloseEvent: boolean
+  setShouldLogModalCloseEvent: (shouldLog: boolean) => void
+  showAcceptChanges: boolean
+  onAcceptChanges: () => void
 }) {
   const transactionDeadlineSecondsSinceEpoch = useTransactionDeadline()?.toNumber() // in seconds since epoch
   const isAutoSlippage = useUserSlippageTolerance()[0] === 'auto'
   const [routerPreference] = useRouterPreference()
-  const routes = getRoutingDiagramEntries(trade)
+  const routes = trade ? getRoutingDiagramEntries(trade) : []
+  const [lastExecutionPrice, setLastExecutionPrice] = useState(trade?.executionPrice)
+  const [priceUpdate, setPriceUpdate] = useState<number>()
+  const theme = useTheme()
+  const { chainId } = useWeb3React()
+  const nativeCurrency = useNativeCurrency(chainId)
+
+  useEffect(() => {
+    if (trade && lastExecutionPrice && !trade?.executionPrice.equalTo(lastExecutionPrice)) {
+      setPriceUpdate(getPriceUpdateBasisPoints(lastExecutionPrice, trade?.executionPrice))
+      setLastExecutionPrice(trade?.executionPrice)
+    }
+  }, [lastExecutionPrice, setLastExecutionPrice, trade])
+
+  useEffect(() => {
+    if (trade && shouldLogModalCloseEvent && showAcceptChanges) {
+      sendAnalyticsEvent(
+        SwapEventName.SWAP_PRICE_UPDATE_ACKNOWLEDGED,
+        formatSwapPriceUpdatedEventProperties(trade, priceUpdate, SwapPriceUpdateUserResponse.REJECTED)
+      )
+    }
+    setShouldLogModalCloseEvent(false)
+  }, [shouldLogModalCloseEvent, showAcceptChanges, setShouldLogModalCloseEvent, trade, priceUpdate])
+
+  const details = useMemo(() => {
+    const label = `${trade?.executionPrice.baseCurrency?.symbol} `
+    const labelInverted = `${trade?.executionPrice.quoteCurrency?.symbol}`
+    const formattedPrice = formatTransactionAmount(priceToPreciseFloat(trade?.executionPrice))
+    const details: Array<DetailItem> = [
+      { label: t`Exchange rate`, value: `${'1 ' + labelInverted + ' = ' + formattedPrice ?? '-'} ${label}` },
+      {
+        label: t`Network fee`,
+        value: `~${formatCurrencyAmount(trade?.gasUseEstimateUSD, NumberType.FiatGasPrice)}`,
```

Comment:
When converting to a static JSX content, please make sure to not call `formatCurrencyAmount` when `gasUseEstimateUSD` is false.

Reply:
done